### PR TITLE
[T5A1][W13-B1] Song Zhiwen

### PR DESCRIPTION
**Pull request by zzzzwen**

### CODE REVIEW

File: doc/DeveloperGuide.md [hunk 1]
```diff
@@ -43,10 +43,15 @@ Priority | As a ... | I want to ... | So that I can...
 -------- | :-------- | :--------- | :-----------
 `* * *` | new user | see usage instructions | refer to instructions when I forget how to use the App
 `* * *` | user | add a new person |
+`* * *` | user | edit an existing person | update the person's contact information
 `* * *` | user | delete a person | remove entries that I no longer need
 `* * *` | user | find a person by name | locate details of persons without having to go through the entire list
 `* *` | user | hide [private contact details](#private-contact-detail) by default | to minimize chance of someone else seeing them by accident
+`* *` | user | reveal contact details | to allow someone to see the necessary contacts
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: doesn't the view function already do that?

File: doc/DeveloperGuide.md [hunk 2]
```diff
@@ -74,12 +79,46 @@ Use case ends.
 > 3a1. AddressBook shows an error message <br>
   Use case resumes at step 2
 
+#### Use case: Edit tag
+
+**MSS**
+
+1. User requests to list persons
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: the activity was updated to rename a tag; this means, if the tag is changed from "friend" to "buddies", all people associated with "friend" tag should be updated to have tag as "buddies"

File: doc/DeveloperGuide.md [hunk 3]
```diff
@@ -74,12 +79,46 @@ Use case ends.
...
+2. AddressBook shows a list of persons
+3. User requests to edit the tag of a specific person in the list
+4. AddressBook lists all the tags of selected person
+5. User inputs the tag to be changed
+6. AddressBook shows the details of the tag
+7. User inputs the new tag name
+8. AddressBook requests for confirmation to change
+9. User confirms the change
+10. AddressBook changes the tag with a new tag name. 
+Use case ends.
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: you have got "how to write the use case?" right :+1:

File: doc/DeveloperGuide.md [hunk 4]
```diff
@@ -74,12 +79,46 @@ Use case ends.
...
+> 5a1. AddressBook shows an error message, requesting for another tag name
+  Use case resumes at step 4
+
 ## Appendix C : Non Functional Requirements
 
 1. Should work on any [mainstream OS](#mainstream-os) as long as it has Java 8 or higher installed.
 2. Should be able to hold up to 1000 persons.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
+5. Should have a clear UI
```
Comment (okkhoy) [inline doc/DeveloperGuide.md]: what does clear mean?
- In general it is better to be as specific and objective as possible.
- words like less, more, fast, etc., are subjective

**Comment by okkhoy** [comment]
@zzzzwen 
Some comments added. Please ack & close the PR after reading comments.

**Comment by zzzzwen** [comment]
noted

**Pull request closed without merging**